designer/server/src/main/scala/pl/touk/nussknacker/ui/util/GenerateDesignerOpenApiYamlFile.scala

```diff
@@ -0,0 +1,21 @@
+package pl.touk.nussknacker.ui.util
+
+import better.files.Dsl._
+import cats.effect.{ExitCode, IO, IOApp}
+import com.typesafe.scalalogging.StrictLogging
+import pl.touk.nussknacker.ui.services.NuDesignerApiAvailableToExpose
+
+object GenerateDesignerOpenApiYamlFile extends IOApp with StrictLogging {
```

Comment:
I thought about having this generate logic in the same place as the test is. WDYT? We have similar approach in `ClassExtractionBaseTest` which is a base class for two test cases (`DefaultClassExtractionTest` and `DevClassExtractionTest`) We check there how changes in configuration of "model" and in definition of our component have influence on classes and methods that are available in spel expressions.

Reply:
This file has two purposes:
* it's being run by the `GenerateDesignerOpenApiPlugin`
* it can be run by dev manually

I agree that the class should not be a part of the final jar. But at the same, it's not a test source file. This class is for a developer. They should take advantage of it to generate OpenAPI that can be easily reviewed in terms of backward compatibility in the review process (IMO it's easier to find it by reading the OpenAPI file than looking at Tapir's code). At the same time, the file is helpful when you want to check what API designer has (eg. for the frontend team). All of these are development-related activities. 

I think it'd be nice to have a module for development tools. Like this one. So, this is the direction I'd like to go if the current approach is not acceptable for now.

Reply:
> it's being run by the GenerateDesignerOpenApiPlugin
This plugin will be run by whom? Not a developer?

Writing tests is not a development-related activity? What about other places where we already have "some" approach like this `DefaultClassExtractionTest`? Don't you think that there is some value in having things consistent?

Reply:
Obviously, it's. Writing and reading tests are a part of the developer's job. It's a special place for tests in a project (it's a `test` folder obviously). But there are some other things that a developer is supposed to do. One of the things is a review. Sometimes, there are some things in the project that are useful for other devs but are not production code or tests. This is the thing. 

So in this particular case, we have:
* `NuDesignerApiAvailableToExpose` - this is the object that has all the necessary logic to generate OpenAPI yaml for all endpoints defined in the `BaseEndpointDefinitions` class
* `GenerateDesignerOpenApiYamlFile` - is an object with the main method to generate the file (it's using the above one)
* `NuDesignerApiAvailableToExposeYamlSpec` - it a test if the generated OpenAPI yaml is a fresh one
* `GenerateDesignerOpenApiPlugin` - it's a plugin that runs the main method of the GenerateDesignerOpenApiYamlFile

The last one has to stay where it is now. The 3rd one is in tests - the place it is supposed to be. 
The first two, are in the "main" code. I agree it's not a proper place for these files. 
I could obviously move them to the test code but I'm not sure if this is the right approach. I proposed a new module (with dev tools).

I (as you probably know, because I used to mention that) see value in having things consistent, but I also like to improve things. IMO, this is sth we can easily improve. 

--- 
> We check there how changes in configuration of "model" and in definition of our component have influence on classes and methods that are available in spel expressions.

As I described above we don't want to only test sth in this case. We want to give developers a tool to generate static OpenAPI file. I'm not sure I see that the case you show me is the same. If I'm wrong, please explain (because I found the tests are not self-explanatory in this matter).

Reply:
>  I'm not sure I see that the case you show me is the same. If I'm wrong, please explain

It is exactly the same as with OpenAPI. We have some JSON format that describes the API that we expose to our users in SPeL. We have our own mechanism of automatic derivation of this API based on components, helpers (utils like `#COLLECTION` variable) and configuration (e.g. dev-application.conf). 

I appreciate your spirit to improve things but I was worried that we will have two similar approaches which two other concepts of solving this trade-off between a number of modules, having things separated etc. Let's extract this model and at least add a TODO into these two classes (`DefaultClassExtractionTest` and `DevClassExtractionTest`) that we want to follow this approach in this place as well.

Reply:
I've moved the 1st and the 2nd class to the test sources, because I want to close this task as soon as I can. I'll create a jira for the improvement. 

Please take a look if it can be done like this, for now.